packages/augur-state/src/db/DB.ts

```diff
@@ -0,0 +1,51 @@
+import { SyncableDB } from './SyncableDB';
+import { Augur } from 'augur-api';
+import { SyncStatus } from './SyncStatus';
+import { TrackedUsers } from './TrackedUsers';
+import { UserSyncableDB } from './UserSyncableDB';
+
+export class DB<TBigNumber> {
+  private syncableDatabases: Array<SyncableDB<TBigNumber>> = [];
+
+  public syncStatus: SyncStatus;
+  public trackedUsers: TrackedUsers;
+  public marketCreated: SyncableDB<TBigNumber>;
+  public marketFinalized: SyncableDB<TBigNumber>;
+
+  public userTokensMinted: { [user: string]: UserSyncableDB<TBigNumber>} = {};
+
+  public constructor () {}
+
+  public static async createAndInitializeDB<TBigNumber>(): Promise<DB<TBigNumber>> {
+    const dbController = new DB<TBigNumber>();
+    await dbController.initializeDB();
+    return dbController;
+  }
+
+  public async initializeDB(): Promise<void> {
+    this.syncStatus = new SyncStatus();
+    this.trackedUsers = new TrackedUsers();
```

Comment:
This is the code that keeps track of which users we need to sync the trade history for?